File: database-tests/app/src/RepositoryFactory/RepositoryFactoryMongoDB.py
from src.data.Frame.IFrameRepositoryKeyValue import IFrameRepositoryKeyValue
from src.data.Frame.IFrameRepositoryDocument import IFrameRepositoryDocument
from .RepositoryAbstractFactory import RepositoryAbstractFactory
from src.data.Frame.FrameRepositoryMongoDB.FrameRepositoryMongoDBKeyValue import FrameRepositoryMongoDBKeyValue
from src.data.Frame.FrameRepositoryMongoDB.FrameRepositoryMongoDBDocument import FrameRepositoryMongoDBDocument

class RepositoryFactoryMongoDB(RepositoryAbstractFactory):
  def createFrameRepositoryKeyValue(self) -> IFrameRepositoryKeyValue:
    return FrameRepositoryMongoDBKeyValue()
  
  def createFrameRepositoryDocument(self) -> IFrameRepositoryDocument:
    return FrameRepositoryMongoDBDocument()

File: vision-be/vision-panettone/vision/RepositoryFactory/RepositoryFactoryMongoDB/RepositoryFactoryMongoDB.cpp
// RepositoryFactoryMongoDB.cpp
#include "RepositoryFactoryMongoDB.h"

#include "../../data/Frame/FrameRepositoryMongoDB/FrameRepositoryMongoDB.h"
#include "../../data/Frame/IFrameRepository.h"

std::unique_ptr<IFrameRepository> RepositoryFactoryMongoDB::createFrameRepository() const {
  return std::make_unique<FrameRepositoryMongoDB>();
}
